py/objmodule.c

```diff
@@ -61,7 +61,20 @@ STATIC void module_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
         mp_map_elem_t *elem = mp_map_lookup(&self->globals->map, MP_OBJ_NEW_QSTR(attr), MP_MAP_LOOKUP);
         if (elem != NULL) {
             dest[0] = elem->value;
+        #ifndef MICROPY_CPYTHON_COMPAT
```

Comment:
maybe #if MICROPY_PEP562, though incomplete because it should covers __dir__ too

setattr already works except for C/C++ modules which is normal behaviour afaik.